### PR TITLE
Feature: Create generating invitation link fn

### PR DESCRIPTION
**Pull request by ShoeheyOt**
## Overview

implement fetching hash value

## Screen Captures

[invHash_test.webm](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/4f70ac37-e754-40f6-adb7-b84c04182b17)

## References

- https://dev.to/hirohata/created-result-type-in-typescript-128m
- https://jestjs.io/docs/expect

## Notes

-

## Assignee Checklist:

- [X] The naming convention of the PR title is correct (See the comment at the top of this template)
- [X] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [X] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [X] The correct assignees and reviewers have been designated for this PR
- [X] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [X] All the related issues are associated with this PR
- [X] All criteria in the associated issues are met (please tick the checkboxes)
- [X] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [x] Double-check the "Assignee Checklist"
- [x] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] The code is well-commented and easy to understand
- [x] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/features/groups/components/InviteMemberDialog/InviteMemberDialogContent.tsx [hunk 1]
```diff
@@ -7,9 +7,10 @@ import {
   DialogTitle,
   Icon,
 } from '@/components/ui';
+import { generateInvitationLinkHash } from '@/lib/api/group/client/groupApiClient.client';
 
 import { useEffect, useState } from 'react';
-
+const BACKEND_API_DOMAIN = process.env.NEXT_PUBLIC_BACKEND_API_DOMAIN || '';
```
Comment (nick-y-ito) [inline src/features/groups/components/InviteMemberDialog/InviteMemberDialogContent.tsx:13]: Please add a line break before each function declaration. If it has a comment block, then the line break should be right before the comment.

File: src/features/groups/components/InviteMemberDialog/InviteMemberDialogContent.tsx [hunk 2]
```diff
@@ -19,14 +20,27 @@ import { useEffect, useState } from 'react';
  * @param props.isDialogOpen - state to control the visibility of dialog, if it is open => true, if not => false, this state is used to switch the text in button.
  * @returns - The JSX code for rendering the dialog component.
  */
-export const InviteMemberDialogContent = ({ isDialogOpen }: { isDialogOpen: boolean }) => {
+export const InviteMemberDialogContent = ({
+  isDialogOpen,
+  groupId,
```
Comment (nick-y-ito) [inline src/features/groups/components/InviteMemberDialog/InviteMemberDialogContent.tsx:25]: `@param` for groupId is missing

File: src/components/page/MembersPage.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ interface IMembersPageProps {
   id: string;
 }
 
-export const MembersPage = async ({ id }: IMembersPageProps) => {
+export const MembersPage: React.FC<IMembersPageProps> = async ({ id }) => {
```
Comment (nick-y-ito) [inline src/components/page/MembersPage.tsx:11]: Please avoid using wildcard imports.
See: https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md#avoid-using-wildcard-imports

File: src/lib/api/group/client/groupApiClient.client.ts [hunk 4]
```diff
@@ -77,3 +77,34 @@ export const putRenameGroup = async (
...
+}
+
+/**
+ * generate a invitation link hash
+ * @param groupId unique Id for generating invitation link hash
+ * @returns A hash: IGenerateInvitationLink of inivitation link
+ */
+export const generateInvitationLinkHash = async (
+  groupId: string,
+): Promise<Result<string, string>> => {
```
Comment (nick-y-ito) [inline src/lib/api/group/client/groupApiClient.client.ts:96]: Currently, it is not very clear what the type `string` refers to for the return type:
```ts
Result<string, string>
```

I suggest creating an interface for the return value and changing the return type like so:
```ts
interface IPutgenerateInvitationLinkHashResponse {
  hash: string
  // This structure allows to add more values for the future
  // It's also kind of a convention to always make the return value an object
}
:
Promise<Result<IPutgenerateInvitationLinkHashResponse, string>>
```

File: src/lib/api/group/client/groupApiClient.client.ts [hunk 5]
```diff
@@ -77,3 +77,34 @@ export const putRenameGroup = async (
...
+export interface IGenerateInvitationLink {
+  invitationLinkHash: string;
+}
+
+/**
+ * generate a invitation link hash
+ * @param groupId unique Id for generating invitation link hash
+ * @returns A hash: IGenerateInvitationLink of inivitation link
+ */
+export const generateInvitationLinkHash = async (
```
Comment (nick-y-ito) [inline src/lib/api/group/client/groupApiClient.client.ts:94]: As per convention, please rename it to start with `put` (one of the request methods).

**Review by nick-y-ito — CHANGES_REQUESTED** [review]
@ShoeheyOt 
Thank you for your work! Please take a look at the comments below and address those issues :)

**Comment by kotaaaa** [comment]
[This button](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/src/features/groups/components/MemberList.tsx#L14-L16) at GroupSingle page should have same functionality of [InviteMemberDialogContent.tsx](https://github.com/nishiki-tech/nishiki-frontend/pull/245/files#diff-989ac945aae3710b341f62402e478ce243d3c632b0135823e1046fb4d7270e86).

**Review by kotaaaa — APPROVED** [review]

**Review by nick-y-ito — APPROVED** [review]
@ShoeheyOt
Thank you for fixing them! I think it's good to go 😄

**Pull request merged**